### ISSUE TITLE
Error importing matplotlib: No module named 'tornado'

### ISSUE THREAD

**Issue by DavidA2014** [pyinstaller/pyinstaller#1867]
Hi

I am running pyinstaller dev version on Fedora 22. I am trying to package a Kivy application that uses matplotlib.

I installed pyinstaller as follows:

```sudo -E pip3 install https://github.com/pyinstaller/pyinstaller/archive/develop.zip```

I installed Kivy by cloning the Kivy repository, building and installing it:

```
git clone https://github.com/kivy/kivy.git
cd kivy
python3 setup.py build_ext --inplace -f
sudo python3 setup.py install
```

I installed matplotlib using pip3:

```
sudo -E pip3 install --upgrade setuptools
sudo -E yum -y install freetype-devel
sudo -E pip3 install matplotlib
```

I installed the Kivy matplotlib backend using pip3 and Kivy garden:

```
sudo -E pip3 install kivy-garden
garden install matplotlib
```

My app runs fine from the installed locations.

I then run pyinstaller on it as follows:

```
$ pyinstaller --name gui_dist ~/trunk/
[pyinstaller_op.txt](https://github.com/pyinstaller/pyinstaller/files/165098/pyinstaller_op.txt)
gui/main.py
```

I see lots of warnings and errors regarding matplotlib, specifically:

```
17478 INFO:   Matplotlib backend "WebAgg": ignored
    Traceback (most recent call last):
  File "/usr/lib64/python3.4/site-packages/matplotlib/backends/backend_webagg.py", line 28, in <module>
    import tornado
ImportError: No module named 'tornado'
```

(I have attached the entire output as a separate file).

My problem seems similar to [this issue](https://github.com/pyinstaller/pyinstaller/issues/1190) which says it's fixed by [this pull request](https://github.com/pyinstaller/pyinstaller/pull/1194) so I hoped it was fixed (I'm using the latest development version).

Please can you advise me how to fix this problem?

Best regards

David
[pyinstaller_op.txt](https://github.com/pyinstaller/pyinstaller/files/165135/pyinstaller_op.txt)

**Comment by gdementen** [comment]
AFAIK, the "tornado" error is harmless. I also got that and my executables work fine. Your problem seems to be with gdk stuff:
```
 FileNotFoundError: [Errno 2] No such file or directory: 'gdk-pixbuf-query-loaders'
```

**Comment by leycec** [comment]
**Interesting.**

As gdementen accurately notes _and_ as the `Matplotlib backend "WebAgg": ignored` line suggests, the `tornado` warning is non-fatal and hence ignorable. Arguably, we could improve the human readability of exceptions raised by Matplotlib backend detection. (Any volunteers? _Anybody?_ **_Hello?_**)

The `gdk-pixbuf-query-loaders` exception, however, is fatal and hence _not_ ignorable. That's _not_ good. It's also a bit of an edge case. Our `hook-gi.repository.GdkPixbuf` hook internally calls the external `gdk-pixbuf-query-loaders` command. This hook is run _only_ if the `gi.repository.GdkPixbuf` module is importable, which should only be the case if pyGObject >= 2.0 is installed, which should only be the case if both GTK+ _and_ GDK are installed, in which case `gdk-pixbuf-query-loaders` should also be installed.

The prior paragraph contains an absurd number of "should"s. Clearly, one of these "should"s breaks down in the real world. I have an [open pull request](https://github.com/pyinstaller/pyinstaller/pull/1843) addressing `hook-gi.repository.GdkPixbuf`, so I'll go ahead and address this issue as well.

Thanks for the detailed issue breakdowns, everybody!

**Comment by leycec** [comment]
**Fixed by #1843.** Thanks again, all.
